test/analyzer_test.rb

```diff
@@ -1,16 +1,22 @@
 require "test_helper"
 
 class AnalyzerTest < Minitest::Test
-  def test_that_it_converts_to_path_and_runs
-    slug = mock
-    solution_path = mock
-    output_path = mock
-    solution_pathname = mock
-    output_pathname = mock
-    Pathname.expects(:new).with(solution_path).returns(solution_pathname)
-    Pathname.expects(:new).with(output_path).returns(output_pathname)
+  TMP_PATH = Pathname.new('/tmp')
 
-    AnalyzeSolution.expects(:call).with(slug, solution_pathname, output_pathname)
-    Analyzer.analyze(slug, solution_path, output_path)
+  protected
+  attr_reader :exercise_slug
+
+  def analysis_results(code = "")
```

Comment:
I created a parent class with a method that is called by all of the existing tests. The new method creates a temporary file and then calls the actually AnalyzeSolution class to read and parse the file.